### ISSUE TITLE
Experiment on a Low-resource Language Pair

### ISSUE THREAD

**Issue by Surnal** [kenji-imamura/recycle_bert#2]
Hello, I'd like to ask about the low-resource translation task of en-vi. 
1. When you use sentencepiece to divide sub-words, what is the vocab size? 
2. When training, what is the size of max-tokens? What is UPDATES_PER_EPOCH?
3. What does the scoring tool use？
If so, is it convenient to provide training scripts for low-resource translation tasks?
Thank you very much for your reply.

@kenji-imamura

**Comment by kenji-imamura** [comment]
Hi, I answer your questions.

1. The vocabulary size for Vietnamese was 16K.
Note that the English vocabulary size was 30K according to the BERT tokenizer.

2. In my training, I set the max-tokens for Vi 6,000 (tokens).

The UPDATES_PER_EPOCH constant indicates the number of updates in an epoch.  This constant is used for converting the number of epochs into the number of updates because I want to warm-up five epochs in training.  In my case, the En-Vi corpus was around 130K sentences and the mini-batch size was around 500 sentences.  Therefore, I set the constant 260 (=130K/500).

3. I used multi-bleu.pl in the Moses toolkit for the evaluation.
(The MultEval tool written in my paper can also be used.)

**Comment by Surnal** [comment]
> Hi, I answer your questions.
> 
> 1. The vocabulary size for Vietnamese was 16K.
>    Note that the English vocabulary size was 30K according to the BERT tokenizer.
> 2. In my training, I set the max-tokens for Vi 6,000 (tokens).
> 
> The UPDATES_PER_EPOCH constant indicates the number of updates in an epoch. This constant is used for converting the number of epochs into the number of updates because I want to warm-up five epochs in training. In my case, the En-Vi corpus was around 130K sentences and the mini-batch size was around 500 sentences. Therefore, I set the constant 260 (=130K/500).
> 
> 1. I used multi-bleu.pl in the Moses toolkit for the evaluation.
>    (The MultEval tool written in my paper can also be used.)


Thank you very much for your reply. 
I have one more question to ask you. How do you calculate the mini-batch size? I use the fairseq toolkit, do I just need to look at the bsz= on the printed log and control it to around 500?

**Comment by kenji-imamura** [comment]
I set the mini-batch size by trying training to make the number of updates per epoch around 260.
